src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -901,10 +902,34 @@ protected virtual void Dispose(bool disposing)
 
         public void Dispose()
         {
+            StopWorkerProcess();
             _disposing = true;
             Dispose(true);
         }
 
+        private void StopWorkerProcess()
+        {
+            bool capabilityEnabled = !string.IsNullOrEmpty(_workerCapabilities.GetCapabilityState(RpcWorkerConstants.HandlesWorkerTerminateMessage));
```

Comment:
Is there a possibility that this could evaluate to not empty but have the value "false"?

Reply:
I don't think so, my understanding of capabilities is that if it is in the list then we treat it as "true" / something the worker can handle - the value provided doesn't matter.

E.g. if a worker added the capability `HandlesWorkerTerminateMessage = false` to their init response message, we don't actual care about the value provided, we only check that `HandlesWorkerTerminateMessage` [is in the capability list provided by the worker](https://github.com/Azure/azure-functions-host/blob/5a79ab338e0e6ebd961a30b9950fc19ac9645fa0/src/WebJobs.Script.Grpc/Channel/GrpcCapabilities.cs#L23)